framework/include/reporters/ReporterContext.h

```diff
@@ -421,16 +458,97 @@ class ReporterGeneralContext : public ReporterContext<T>
                                   const MooseObject & producer) const override;
 
   virtual void resize(dof_id_type local_size) final;
+  virtual void clear() final;
+  virtual void vectorSum() final;
 
   virtual std::string contextType() const override { return MooseUtils::prettyCppType(this); }
 };
 
+//  Needed for compile-time checking if T is a vector.
+template <typename>
+struct is_std_vector : std::false_type
+{
+};
+
+template <typename T, typename A>
+struct is_std_vector<std::vector<T, A>> : std::true_type
+{
+};
+
 template <typename T>
-void ReporterGeneralContext<T>::resize(dof_id_type)
+void
+ReporterGeneralContext<T>::resize(dof_id_type size)
 {
-  mooseError("Cannot resize non vector-type reporter values.");
+  if constexpr (is_std_vector<T>::value)
+    this->_state.value().resize(size);
+  else
+  {
+    libmesh_ignore(size);
+    mooseError("Cannot resize non vector-type reporter values.");
+  }
 }
+template <typename T>
+void
+ReporterGeneralContext<T>::clear()
+{
+  if constexpr (is_std_vector<T>::value)
+    this->_state.value().clear();
+  else
+    mooseError("Cannot clear non vector-type reporter values.");
+}
+template <typename T>
+void
+ReporterGeneralContext<T>::vectorSum()
+{
+  // Case 1: T is a numeric type that we can sum (excluding bool)
+  if constexpr (std::is_arithmetic<T>::value && !std::is_same<T, bool>::value)
+  {
+    // Perform summation of the scalar value across all processors
+    this->comm().sum(this->_state.value());
+    return;
+  }
+  // Case 2: T is a vector type
+  else if constexpr (is_std_vector<T>::value)
+  {
+    using VectorValueType = typename T::value_type;
+
+    // Check if the vector elements are of a numeric type
+    if constexpr (std::is_arithmetic<VectorValueType>::value &&
+                  !std::is_same<VectorValueType, bool>::value)
+    {
+      // Perform summation of the vector elements across all processors
+      this->comm().sum(this->_state.value());
+      return;
+    }
+    // Check if the vector elements are also vectors
+    else if constexpr (is_std_vector<VectorValueType>::value)
+    {
+      using InnerValueType = typename VectorValueType::value_type;
+
+      // Check if the inner vector elements are of a numeric type
+      if constexpr (std::is_arithmetic<InnerValueType>::value &&
+                    !std::is_same<InnerValueType, bool>::value)
```

Comment:
you have to error on the 'else' case since you are not doing anything

Reply:
If a code never performs a vector sum they will hit the moose error at the bottom of the function. Every sum has a return right after it.